src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,522 @@
+            continue
+
+        file.write(f"  - name: {interface.name}\n")
+        file.write("    properties:\n")
+
+        for prop_key, prop_value in interface.properties.items():
+            file.write(f"      - name: {prop_key}\n")
+            if prop_value.description is not None:
+                file.write(f"        description: \"{prop_value.description}\"\n")
+            file.write(f"        type: {prop_value.type.type}\n")
+            if prop_value.type.is_array:
+                file.write("        array: true\n")
+
+    if len(module.enumerations) > 0:
+        file.write("\n")
+        file.write("enums:\n")
+
+        for enumeration in module.enumerations:
+            file.write(f"  - name: {enumeration.name}\n")
+            file.write("    members:\n")
+
+            for variant, value in enumeration.variants:
+                file.write(f"      - name: {variant}\n")
+                if value is not None:
+                    file.write(f"        value: {value}\n")
+
+    if len(module.structures) > 0:
+        file.write("\n")
+        file.write("structs:\n")
+
+        for structure in module.structures:
+            file.write(f"  - name: {structure.name}\n")
+            file.write("    fields:\n")
+
+            for prop_key, prop_value in structure.properties.items():
+                file.write(f"      - name: {prop_key}\n")
+                if prop_value.description is not None:
+                    file.write(f"        description: \"{prop_value.description}\"\n")
+                file.write(f"        type: {prop_value.type.type}\n")
+                if prop_value.type.is_array:
+                    file.write("        array: true\n")
+
+
+def export_apigear(directory: Path, root: VSSNode, data_type_tree: VSSNode | None,
+                   layers: typing.Dict[SolutionLayers, Path]):
+    """This method is used to traverse through the root VSS node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a file
+    """
+    module = ApiGearModule()
+    module_name = root.name
+    module_filename = MODULE_FILENAME.format(module_name)
+    module_path = os.path.join(directory, module_filename)
+
+    log.info(f"directory: {directory}")
+    log.info(f"Module name: {module_name}")
+    log.info(f"Module filename: {module_filename}")
+    log.info(f"Module Path: {module_path}")
+
+    if (not layers):
+        layers = {SolutionLayers.CPP: Path("../cppservice")}
+        log.warning(f"No layers provided! Defaulting to CPP in {layers[SolutionLayers.CPP]}")
+
+    log.debug(layers)
+
+    export_node(root, module)
+    export_data_type_node(data_type_tree, module)
+    generate_solution(directory, module_filename, module_name, layers)
+    with open(module_path, "w") as file:
+        generate_module(file, module, module_name)
+        log.info(f"Generated module file at {module_path}")
+
+    log.info("Apigear files generated at location : " + file.name)
```

Comment:
file.name probably does not reveal a good location. Just `file` probably does

Reply:
Properly used the `directory` variable here.